rfcs/0000-kuadrant-release-process.md

```diff
@@ -0,0 +1,207 @@
+is ready, and it's been tested by the Engineering team. The QA team should be notified in advance of the release, and
+they should be able to assert the release candidate is ready before the actual release happens.
+
+There is an ideal time to hand over to the QA team for testing, especially since we are using GitHub for orchestration,
+we could briefly define it in the following steps:
+
+1. Complete Development Work: The engineering team completes their work included in the milestone.
+2. Create Release Candidate: The engineering team creates Release Candidate builds and manifests for all components required
+for the release
+3. Notify QA Team: At this time, the engineering team can notify the QA team that the deliverables for the release candidate
+are ready for testing. This notification can be done through the GitHub Kuadrant board, or simply a public message in the
+Kuadrant Slack channel.
+4. Flagging/Testing: The QA team do the actual assertion/testing of the release candidate, checking for any obvious bugs or issues.
+Then QA reports all the bugs as GitHub issues and communicates testing status back publicly on Slack and/or email.
+5. Iterate: Based on the feedback from the QA team, the development team makes any necessary adjustments and repeats the
+process until the release candidate is deemed ready for production.
+6. Publish Release: Once QA communicates that the testing has been successfully finished, the engineering team will publish
+the release both on Github and in the corresponding registries, updates documentation for the new release, and communicates
+it to all channels specified in Communication section.
+
+## Cadence
+
+Once the project is stable enough, and it's adoption increases, the community will be expecting a certain degree of
+commitment from the maintainers, and that includes a regular release cadence. The frequency of the releases of the
+different components could vary depending on the particular component needs. However, the **Kuadrant Operator**
+it's been discussed in the past that it should be released every 3-4 weeks initially, including the latest released version
+of every component in the suite. There's another RFC that focuses on the actual frequency of each component, one could
+refer to the [Kuadrant Release Cadence RFC](https://github.com/pmccarthy/architecture/blob/release-cadence-proposal/rfcs/0007-project-release-cadence.md).
+
+There are a few reasons for this:
+
+- Delivering Unparalleled Value to Customers: Regular releases can provide customers with regular updates and improvements.
+  These updates can include new features and essential bug fixes, thus enhancing the overall value delivered to the customers.
+- Maximizing Deployment Efficiency: By releasing software at regular intervals, teams can align their activities with
+  available resources and environments, ensuring optimal utilization. This leads to increased efficiency in the deployment process.
```

Comment:
Is this about cross product/project/dependency compatibility? What "available resources and environments" are we talking about? Or maybe it's not clear to me who the "teams" are in this case.

Reply:
In this case teams would refer to (as of today) Engineering and QA, and in the future probably include Product.  Of course this makes more sense when talking about Downstream, but I think it could apply to Upstream too.

Reply:
I think I know what mean. I first read this from a different angle, of users who have Kuadrant in their CI pipelines.

If it's about Kuadrant teams, "deployment" seems a little out of place IMO, but maybe it's me.

Perhaps "Resource Optimization"? As in teams working together and in sync can plan their resources better (e.g. avoid idle servers, etc?)

Could it be overlapping with _Synchronization_ a little?